Src/Orbiter/Psys.cpp

```diff
@@ -45,42 +35,34 @@ PlanetarySystem::~PlanetarySystem ()
 
 void PlanetarySystem::Clear ()
 {
-	int i, j;
-	DWORD k;
-
 	DestroyDeviceObjects ();
-	if (name) {
-		delete []name;
-		name = 0;
-	}
-	while (nvessel) DelVessel (vessel[0], 0);
-	if (nbody) {
-		for (k = 0; k < nbody; k++) delete body[k]; // delete actual objects
-		delete []body; // delete list
-		body = NULL;
-		nbody = 0;
-	}
-	if (nstar) {
-		delete []star;
-		star = NULL;
-		nstar = 0;
-	}
-	if (nplanet) {
-		delete []planet;
-		planet = NULL;
-		nplanet = 0;
-	}
-	if (ngrav) {
-		delete []grav;
-		grav = NULL;
-		ngrav = 0;
+	m_Name.clear();
+
+	//Vessel destructor broadcasts messages to every other vessel in 'vessels'.
+	//We remove it from the collection as soon as we deleted it to prevent the next Vessel to broadcast to the free'd one.
+	for (size_t i = vessels.size() - 1; i != static_cast<size_t>(-1); --i) { //Special reverse loop to prevent unsigned underflow
```

Comment:
Wouldn't this be even nicer?
```cpp
for (auto it = vessels.rbegin(); it != vessels.rend(); ++it) { // <= reverse iterator ...
	DelVessel(*it);
	vessels.pop_back();
}
```

Reply:
It might be, however I can't guarantee if erasing items (pop_back) during reverse iteration will invalidate the iterator.
It SHOULD be fine, considering that the iterator is probably just an index internally but I assume that the standard makes no guarantees in that regard so I wouldn't take my chances.

Reply:
I am pretty sure the reverse iterators work fine. At least when I test them ( https://godbolt.org/z/dEja7ceTv ) 
I used gcc though, but I bet msvc would behave similar 😀 

> [...]Iterators and references to the last element, as well as the [end()](https://en.cppreference.com/w/cpp/container/vector/end) iterator, are invalidated.[...]

from https://en.cppreference.com/w/cpp/container/vector/pop_back

On a side-note: I really love the cleanup/refactor of the code! Unfortunatly pull-requests are not pulled very frequent 😞

Reply:
> Wouldn't this be even nicer?

@schnepe2 it wouldn't. According to [cppreference.com](https://en.cppreference.com/w/cpp/container/vector/pop_back) calling `pop_back()` invalidates iterator to the last element (ie, `rbegin()`) and the `end()` iterator.

This, however, might be nicer:
```c++
while (vessels.size()) {
    DelVessel(vessels.back());
    vessels.pop_back();
}
```

Reply:
I am very sure you have tested this much more than I did =D 👍